DetectorDescription/DDCMS/src/DDFilteredView.cc

```diff
@@ -560,6 +560,42 @@ DDFilteredView::nav_type DDFilteredView::navPos() const {
   return pos;
 }
 
+const int DDFilteredView::level() const {
+  int level(0);
+  if (not it_.empty()) {
+    level = it_.back().GetLevel();
+  }
+  return level;
+}
+
+bool DDFilteredView::goTo(const nav_type& newpos) {
+  bool result(false);
+
+  // save the current position
+  it_.emplace_back(Iterator(it_.back().GetTopVolume()));
+  Node* node = nullptr;
+
+  // try to navigate down to the newpos
+  int level = 1;
+  for (auto const& i : newpos) {
+    it_.back().SetType(0);
+    node = it_.back().Next();
+    for (int j = 1; j <= i; j++) {
+      it_.back().SetType(1);
+      node = it_.back().Next();
+    }
+    level++;
```

Comment:
Something is missing here. `level` is incremented but never checked. Or should `level` be deleted?

Reply:
Yes, thanks! Well spotted.

Reply:
removed